core/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```
@@ -98,46 +162,210 @@ public FieldSortBuilder sortMode(String sortMode) {
+        XContentParser parser = context.parser();
+
+        QueryBuilder nestedFilter = null;
+        String nestedPath = null;
+        Object missing = null;
+        SortOrder order = null;
+        String sortMode = null;
+        String unmappedType = null;
+
+        String currentFieldName = null;
+        XContentParser.Token token;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (token == XContentParser.Token.START_OBJECT) {
+                if (context.parseFieldMatcher().match(currentFieldName, NESTED_FILTER)) {
+                    nestedFilter = context.parseInnerQueryBuilder();
+                }
+            } else if (token.isValue()) {
+                if (context.parseFieldMatcher().match(currentFieldName, NESTED_PATH)) {
+                    nestedPath = parser.text();
+                } else if (context.parseFieldMatcher().match(currentFieldName, MISSING)) {
+                    missing = parser.objectBytes();
+                } else if (context.parseFieldMatcher().match(currentFieldName, ORDER)) {
+                    String sortOrder = parser.text();
+                    if ("asc".equals(sortOrder)) {
+                        order = SortOrder.ASC;
+                    } else if ("desc".equals(sortOrder)) {
+                        order = SortOrder.DESC;
+                    } else {
+                        throw new IllegalStateException("Sort order " + sortOrder + " not supported.");
+                    }
+                } else if (context.parseFieldMatcher().match(currentFieldName, SORT_MODE)) {
+                    sortMode = parser.text();
+                } else if (context.parseFieldMatcher().match(currentFieldName, UNMAPPED_TYPE)) {
+                    unmappedType = parser.text();
+                }
```

Can we also throw an error when falling through until here?